subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/AbstractGroovyCompileAvoidanceIntegrationSpec.groovy

```diff
@@ -215,4 +215,174 @@ public @interface MyAnnotation { }
         executedAndNotSkipped ":b:compileGroovy"
         failure.assertHasCause('Bad AST transformation!')
     }
+
+    def "doesn't recompile when private element of implementation class changes"() {
```

Comment:
Can we maybe leave most of the test shared between Groovy and Java? I.e. the parts where we do the common changes? And then only add an extra test for the constructor changes to Java?

Reply:
Done - also added normal initializer changes to the Java compile avoidance test